priv/static-template/wasm/popcorn.js

```diff
@@ -240,12 +241,39 @@ export class Popcorn {
     this._iframe = null;
     this._awaitedMessage = null;
     this._listenerRef = null;
+    this._logListeners.clear();
     for (const [_id, callData] of this._calls) {
       const durationMs = performance.now() - callData.startTimeMs;
       callData.reject({ error: new PopcornDeinitializedError("Call cancelled due to instance deinit"), durationMs })
     }
   }
 
+  /**
+   * Registers a log listener that will be called when stdout or stderr output is received.
+   * @param {function({stdout?: string, stderr?: string}): void} listener - Callback function
+   */
+  registerLogListener(listener) {
+    this._logListeners.add(listener);
+  }
+
+  /**
+   * Unregisters a previously registered log listener.
+   * @param {function({stdout?: string, stderr?: string}): void} listener - Callback function to remove
```

Comment:
I don't like conflating stdout with stderr, maybe `registerLogListener(callback, "stdout" | "stderr")`? This way we don't need to do checks in React for `output.stdout !== undefined`.